Flagsmith.FlagsmithClient/Identity.cs

```diff
@@ -1,17 +1,22 @@
-using System.Collections.Generic;
+using Flagsmith.Interfaces;
 using Newtonsoft.Json;
+using System.Collections.Generic;
 
 namespace Flagsmith
 {
     [JsonObject(MemberSerialization.OptIn)]
-    public class Identity
+    public class Identity : IIdentity
     {
         [JsonProperty]
         public List<Flag> flags;
 
         [JsonProperty]
         public List<Trait> traits;
 
+        IReadOnlyCollection<IFlag> IIdentity.Flags => flags;
```

Comment:
Apologies, this is likely my limited .NET knowledge but could you please explain what this is doing? Seems like we're perhaps overriding the `Flags` attributes on the `IIdentity` class but is there a reason we need to reference them as `IIdentity.Flags`?

Reply:
Ah, ok, so am I right in thinking that you created the interface with the properties `Flags` and `Traits` to follow naming conventions. To avoid changing too many things, however, you've left the existing `flags` and `traits` fields on the `Identity` class and simply mapped the new `Flags` and `Traits` properties to the existing fields? 

Other than creating confusion, is there any reason why we can't define them like: 

```csharp
IReadOnlyCollection<IFlag> Flags => flags;
```

Sorry if I misunderstood your comment and you already answered this.

Reply:
it must be "public" (implicit implementation) or with "IIdentity." (explicit implementation). if public, it will be second definition for same reason (Flags vs flags), with explicit it wont be known to Identity class.

Reply:
The interface is also created for another reasons. To have immutable access to data in case you want to cache it (user cannot modify what is cached) and to allow different implementations or hide implementation details (like json serialization and so on)

Reply:
Ok, yep, that makes more sense now. Thanks for explaining.